### ISSUE TITLE
[General Issue]: SSL/QUIC Protocol Errors when Accessing Cloudflare Sites through PiHole + PiVPN + Unbound Setup

### ISSUE THREAD

**Issue by johnsonjamees** [pivpn/pivpn#1876]
### In raising this issue I confirm that

- [X] I have read the [documentation](https://docs.pivpn.io)
- [X] I have read and understood the [PiVPN General Guidelines](https://github.com/pivpn/pivpn/#general-guidelines)
- [X] I have read and understood the [PiVPN Troubleshooting Guidelines](https://docs.pivpn.io/faq/#how-do-i-troubleshoot-connection-issues)
- [X] The issue I am reporting isn't a duplicate, see [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed) and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen).
- [X] I have searched for similar issues and solutions
- [X] I can replicate the issue even after a clean OS installation

### Describe the issue

I'm running Pi-hole, PiVPN, and Unbound on an ARM Compute Instance from Oracle Cloud. My client is a Windows machine behind CGNAT. When connected to PiVPN, I'm experiencing issues accessing websites that use Cloudflare DNS. On Chromium-based browsers, these websites return either `ERR_SSL_PROTOCOL_ERROR` or `ERR_QUIC_PROTOCOL_ERRO`. On Firefox and its forks, the same websites return `SSL_ERROR_RX_MALFORMED_SERVER_HELLO`.

Debug logs
1. Pi-hole: https://pb.envs.net/?384e07e675146a58#buZLxqVM92m7w8N84ZWUg24KLEKV1iqamLgEvGHCygC
2. PiVPN: https://pb.envs.net/?219e94fb105f6001#5DroCbEBa95BQhLxKXcg4beGggLkHCRGBcemfF94Twm3
3. Unbound Configuration: https://pb.envs.net/?ad43cd3d1a71d285#JBqkVydBiyixNB4RXQJBfn8KBjBCP1XXBwAWhhaynCCN

### Expected behavior

All websites should load correctly while connected to PiVPN

### Please describe the steps to replicate the issue

1. Open the WireGuard client for Windows and activate the tunnel.
2. Visit any website using Cloudflare DNS, such as:

- [Claude AI](https://claude.ai/)
- [ChatGPT](https://chatgpt.com/)
- [Pluralsight](https://www.pluralsight.com/)

These websites return errors like "ERR_SSL_PROTOCOL_ERROR" or "ERR_QUIC_PROTOCOL_ERROR."

Websites that do not use Cloudflare DNS, such as [GitHub](https://github.com/) or [Reddit](https://www.reddit.com/), load normally as expected.

### Have you taken any steps towards solving your issue?

Configured the DNS server in the WireGuard configuration to use Cloudflare's public DNS resolver, but it did not resolve the issue.

### Where did you run pivpn?

Arm Compute instance from Oracle Cloud

### Please provide your output from `uname -a`

Linux oracle-vps 6.8.0-1015-oracle #15~22.04.1-Ubuntu SMP Wed Oct  9 16:01:35 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux

### Details about Operative System

PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

### Debug output

[pivpn-debug.log](https://github.com/user-attachments/files/17784468/pivpn-debug.log)

**Comment by orazioedoardo** [comment]
How would Cloudflare being their DNS be an issue? Perhaps you mean websites hosted behind Cloudflare? I don't think this is an issue with PiVPN but rather with the Oracle network or your browsers? Are those up to date? Have you modified them? You could also try https://docs.pivpn.io/faq/#tweaking-mtu

**Comment by johnsonjamees** [comment]
I ran `curl -IL claude.ai` on the server and got an HTTP status code 200 in the response, so the website seems to be reachable on their network.

**Comment by johnsonjamees** [comment]
I removed PiVPN, Unbound, and Pi-hole, then reinstalled Pi-hole and PiVPN (without Unbound this time). However, websites that use Cloudflare's nameservers still won't load.

**Comment by orazioedoardo** [comment]
Please address my comment and questions above instead of reinstalling

**Comment by johnsonjamees** [comment]
I tried adjusting the MTU value from 1410 down to 1280, but it didn't resolve the issue. When I attempted to go lower, the WireGuard app for Windows displayed the following error: `Unable to configure adapter network settings`.

**Comment by johnsonjamees** [comment]
> I don't think this is an issue with PiVPN but rather with the Oracle network or your browsers. Are those up to date?

Yes, they are up to date. I don't think my browsers are causing any issues because when I use a Cloudflare WARP tunnel via the WireGuard client (generated using [wgcf](https://github.com/ViRb3/wgcf)), the websites load normally.
